pipeline/coaddition.py

```diff
@@ -424,3 +447,171 @@ def run(self, images):
+        self.pars.add_defaults_to_dict(photo_cal_config)
+        self.photo_cal = PhotCalibrator(**photo_cal_config)
+
+        self.datastore = None  # use this datastore to save the coadd image and all the products
+
+        self.aligned_images = None  # use this to pass in already aligned images
+
+    def parse_inputs(self, *args, **kwargs):
+        """Parse the possible inputs to the run method.
+
+        The possible input types are:
+        - a list of Image objects
+        - start_time + end_time + instrument + filter + section_id + provenance_id + RA + Dec (or target)
+
+        To pass the latter option, must use named parameters.
+        An optional session can be given, either as one of the named
+        or unnamed args, and it will be used throughout the pipeline
+        (and left open at the end).
+
+        The start_time and end_time can be float (interpreted as MJD)
+        or string (interpreted by astropy.time.Time(...)), or None.
+        If end_time is not given, will use current time.
+        If start_time is not given, will use end_time minus the
+        coaddition pipeline parameter date_range.
+        The provenance_ids can be None, which will use the most recent "preprocessing" provenance.
+        Can also provide a list of provenance_ids or a single string.
+        The coordinates can be given as either float (decimal degrees) or strings
+        (sexagesimal hours for RA and degrees for Dec).
+        Can leave coordinates empty and provide a "target" instead (i.e., target
+        will be used as the "field identifier" in the survey).
+
+        In either case, the output is a list of Image objects.
+        Each image is checked to see if it has the related products
+        (SourceList, PSF, WorldCoordinates, ZeroPoint).
+        If not, it will raise an exception. If giving these images directly
+        (i.e., not letting the pipeline load them from DB) the calling scope
+        must make sure to load those products first.
+        """
+        # first parse the session from args and kwargs
+        args, kwargs, session = parse_session(*args, **kwargs)
+
+        if len(args) == 1 and isinstance(args[0], list):
+            args = args[0]  # in case we are given a list of images
+        elif len(args) == 2 and isinstance(args[0], list) and isinstance(args[1], list):
+            if not all([isinstance(im, Image) for im in args[1]]):
+                raise TypeError('When supplying two lists, both must be lists of Image objects. ')
+            self.aligned_images = args[1]
+            args = args[0]
+        # check if any remaining args are not Image objects
+        for arg in args:
+            if not isinstance(arg, Image):
+                raise ValueError('All unnamed arguments must be Image objects. ')
+
+        images = args
+
+        if len(images) > 0:
+            return images  # TODO: maybe instead allow the kwargs to filter down the list of Images?
```

Comment:
It seems a little weird to store the aligned_images in self, but then return images and require the caller to store in self.  Why not just store the images in self.images instead of returning the images, so that the two are treated the same way?

Reply:
Agreed.